src/components/settings/ARDownloadListSettings.js

```diff
@@ -50,23 +46,32 @@ const renderItem = ({ data, index, item, setData }) => (
 );
 
 export const ARDownloadListSettings = () => {
-  const [data, setData] = useState(dummyData);
-  const [loading, setLoading] = useState(false);
+  const { data: staticData, loading, refetch } = useStaticContent({
+    name: 'arDownloadableDataList',
+    type: 'json'
+  });
+
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(loading);
   const [freeSize, setFreeSize] = useState(0);
 
   useEffect(() => {
     checkFreeStorage();
   }, [data]);
 
   useEffect(() => {
-    checkDownloadData();
-  }, []);
+    setData(staticData);
+
+    if (staticData?.length) {
+      checkDownloadData({ data: staticData });
+    }
+  }, [staticData]);
 
-  const checkDownloadData = async () => {
-    setLoading(true);
+  const checkDownloadData = async ({ data }) => {
+    setIsLoading(true);
     const { newDownloadedData } = await checkDownloadedData({ downloadableData: data });
     setData(newDownloadedData);
-    setLoading(false);
+    setIsLoading(false);
```

Comment:
ok so we have `data`,  `staticData`, `downloadableData` and `newDownloadedData`? i do not know what is going on with these.
when calling `checkDownloadData` `staticData` is passed as `data` and inside `checkDownloadedData` is called with passing `data` as `downloadableData`, which then becomes `newDownloadedData`.
so everything is the same value?
also the both methods reads the same.
please think of some simplifying of that naming problem entirely for the whole component. i guess this will be to heavy later for us to understand.

Reply:
I made a change in the other Pull Request about this.

https://github.com/ikuseiGmbH/smart-village-app-app/pull/563/commits/f292bb0d2aa2a5563071f474458cfa938df31271

Reply:
i left a comment in the new PR, where we can go on with the conversation.